File: jdroid-java-googleplay-publisher/src/main/java/com/jdroid/android/googleplay/publisher/TrackReleaseStatus.java
package com.jdroid.android.googleplay.publisher;

public enum TrackReleaseStatus {
	
	COMPLETED("completed"),
	DRAFT("draft"),
	HALTED("halted"),
	IN_PROGRESS("inProgress");
	
	private String key;
	
	TrackReleaseStatus(String key) {
		this.key = key;
	}
	
	public String getKey() {
		return key;
	}
	
	public static TrackReleaseStatus findByKey(String key) {
		for (TrackReleaseStatus each : TrackReleaseStatus.values()) {
			if (each.getKey().equals(key)) {
				return each;
			}
		}
		return null;
	}
}

File: google-play-api-kotlin/src/main/java/com/dipien/google/play/api/TrackReleaseStatus.kt
package com.dipien.google.play.api

enum class TrackReleaseStatus(val key: String) {

    COMPLETED("completed"),
    DRAFT("draft"),
    HALTED("halted"),
    IN_PROGRESS("inProgress");

    companion object {
        fun findByKey(key: String): TrackReleaseStatus? {
            for (each in values()) {
                if (each.key == key) {
                    return each
                }
            }
            return null
        }
    }
}
